src/neo4j_genai/retrievers/text2cypher.py

```diff
@@ -132,15 +130,18 @@ def get_search_results(
         except ValidationError as e:
             raise SearchValidationError(e.errors()) from e
 
-        if not self.custom_prompt:
-            prompt_template = Text2CypherTemplate()
+        prompt_template = Text2CypherTemplate(template=self.custom_prompt)
+
+        if self.custom_prompt is None:
```

Comment:
This if/else block should not be required anymore, isn't it? It's not an issue if have more parameters than placeholders in the Python `format` method, and we can maybe add `kwargs` to each template's `format` method to prevent errors.

Reply:
we can remove the `schema` and `examples` args, and use kwargs to cover everything besides the `query_text` arg. Then we could remove the if/else block.